From 222b1517cd86e5a3eadcfd7787f757fdb76db11b Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 1 Feb 2019 14:04:42 +0100
Subject: [PATCH 163/174] support: Correct error message for
 TEST_COMPARE_STRING

It should say "string", not "blob".

(cherry picked from commit 6175507c06de56e03407004bd2f289ed2cce034d)
---
 ChangeLog                             |  6 ++++++
 support/support_test_compare_string.c |  2 +-
 support/tst-test_compare_string.c     | 12 ++++++------
 3 files changed, 13 insertions(+), 7 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 190b9ea0d0..23b740dc2e 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,9 @@
+2019-02-01  Florian Weimer  <fweimer@redhat.com>
+
+	* support/support_test_compare_string.c
+	(support_test_compare_string): Use "string" in error message.
+	* support/tst-test_compare_string.c (do_test): Adjust.
+
 2018-11-07  Adhemerval Zanella  <adhemerval.zanella@linaro.org>
 
 	* support/support_test_compare_string.c
diff --git a/support/support_test_compare_string.c b/support/support_test_compare_string.c
index a76ba8eda7..b6eaa24570 100644
--- a/support/support_test_compare_string.c
+++ b/support/support_test_compare_string.c
@@ -75,7 +75,7 @@ support_test_compare_string (const char *left, const char *right,
       || memcmp (left, right, left_length) != 0)
     {
       support_record_failure ();
-      printf ("%s:%d: error: blob comparison failed\n", file, line);
+      printf ("%s:%d: error: string comparison failed\n", file, line);
       if (left_length == right_length && right != NULL && left != NULL)
         printf ("  string length: %zu bytes\n", left_length);
       else
diff --git a/support/tst-test_compare_string.c b/support/tst-test_compare_string.c
index 2a4b258587..d3c037a1ec 100644
--- a/support/tst-test_compare_string.c
+++ b/support/tst-test_compare_string.c
@@ -60,22 +60,22 @@ do_test (void)
 
   TEST_VERIFY
     (strcmp (proc.out.buffer,
-"tst-test_compare_string.c:29: error: blob comparison failed\n"
+"tst-test_compare_string.c:29: error: string comparison failed\n"
 "  left string: 0 bytes\n"
 "  right string: NULL\n"
-"tst-test_compare_string.c:30: error: blob comparison failed\n"
+"tst-test_compare_string.c:30: error: string comparison failed\n"
 "  left string: 1 bytes\n"
 "  right string: 0 bytes\n"
 "  left (evaluated from \"X\"):\n"
 "      \"X\"\n"
 "      58\n"
-"tst-test_compare_string.c:31: error: blob comparison failed\n"
+"tst-test_compare_string.c:31: error: string comparison failed\n"
 "  left string: NULL\n"
 "  right string: 1 bytes\n"
 "  right (evaluated from \"X\"):\n"
 "      \"X\"\n"
 "      58\n"
-"tst-test_compare_string.c:32: error: blob comparison failed\n"
+"tst-test_compare_string.c:32: error: string comparison failed\n"
 "  string length: 4 bytes\n"
 "  left (evaluated from \"abcd\"):\n"
 "      \"abcd\"\n"
@@ -83,13 +83,13 @@ do_test (void)
 "  right (evaluated from \"abcD\"):\n"
 "      \"abcD\"\n"
 "      61 62 63 44\n"
-"tst-test_compare_string.c:33: error: blob comparison failed\n"
+"tst-test_compare_string.c:33: error: string comparison failed\n"
 "  left string: 4 bytes\n"
 "  right string: NULL\n"
 "  left (evaluated from \"abcd\"):\n"
 "      \"abcd\"\n"
 "      61 62 63 64\n"
-"tst-test_compare_string.c:34: error: blob comparison failed\n"
+"tst-test_compare_string.c:34: error: string comparison failed\n"
 "  left string: NULL\n"
 "  right string: 4 bytes\n"
 "  right (evaluated from \"abcd\"):\n"
-- 
2.30.2

